### ISSUE TITLE
Changing immutablejs to ramda

### ISSUE THREAD

**Issue by itaied246** [react-boilerplate/react-boilerplate#2312]
I've seen the [video](https://vimeo.com/168648012) explaining the rational behind this boilerplate.
I have also read most of the docs and the tech stack (most of it I already familiar with).

I am a fan of `fp` and pure functions, as I am developing js using `ramda`.
I find it hard and too complex to use `immutablejs` though, and I was wondering if changing the state in the reducers using `ramda` instead can somehow cause problems with the `pure-component` rendering?

**Comment by joshuaalpuerto** [comment]
Hey @itaied246 I'm not sure on w/c part you are having trouble with `ramda` and `immutablejs`. But I'm using `ramda` along with this boilerplate and it seems working nice though.

**Comment by itaied246** [comment]
I want to replace all of the `immutablejs` references by using `ramda` and eventually uninstalling it.

I was wondering if using `ramda` results (which are plain js data structures) is going to mess with react `pure-component` render and updates, as it shallow compare the state / props.

**Comment by joshuaalpuerto** [comment]
>  is going to mess with react pure-component render and updates

No, if you make sure that your data is immutable. You can check the react doc here why it's useful to use something like [immutable data structure](https://reactjs.org/docs/optimizing-performance.html#using-immutable-data-structures)

**Comment by itaied246** [comment]
How does the `shallow` compare works?
It works the same for `json` and immutablejs `map`?

**Comment by joshuaalpuerto** [comment]
> How does the shallow compare works?
It works the same for json and immutablejs map?

You can check the [stack overflow](https://stackoverflow.com/questions/36084515/how-does-shallow-compare-work-in-react?answertab=active#tab-top) answer here.

> Shallow compare does check for equality. When comparing scalar values (numbers, strings) it compares their values. When comparing objects, it does not compare their's attributes - only their references are compared (e.g. "do they point to same object?).
